From a0c840cf0bfbed6e4af5036b808e64fb21e12738 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 16 Jun 2022 09:37:55 +0800
Subject: [PATCH] s32g: aptiv: usb: Use specific TARGET_APTIV_CVC_SOUSA config

Because of SDK commit 0ef56d2b7d ("s32g: evb: Use specific
'S32G2/S32G3-EVB' configs"), we need to add TARGET_APTIV_CVC_SOUSA
config according to TARGET_S32G2XXAEVB.

Upstream-Status: Pending

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/usb/host/ehci-mx6.c | 24 ++++++++++++++++--------
 1 file changed, 16 insertions(+), 8 deletions(-)

diff --git a/drivers/usb/host/ehci-mx6.c b/drivers/usb/host/ehci-mx6.c
index 490b0b7bcf..8afc2db210 100644
--- a/drivers/usb/host/ehci-mx6.c
+++ b/drivers/usb/host/ehci-mx6.c
@@ -148,14 +148,16 @@ static void usb_power_config(int index)
 #endif
 
 #if defined(CONFIG_MX6) || defined(CONFIG_TARGET_S32G2XXAEVB) || \
-	defined(CONFIG_TARGET_S32G3XXAEVB)
+	defined(CONFIG_TARGET_S32G3XXAEVB) || \
+	defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 /* Return 0 : host node, <>0 : device mode */
 static int usb_phy_enable(int index, struct usb_ehci *ehci)
 {
 	int ret;
 	void __iomem *usb_cmd;
 #if !defined(CONFIG_TARGET_S32G2XXAEVB) && \
-	!defined(CONFIG_TARGET_S32G3XXAEVB)
+	!defined(CONFIG_TARGET_S32G3XXAEVB) && \
+	!defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 	void __iomem *phy_reg;
 	void __iomem *phy_ctrl;
 
@@ -179,7 +181,8 @@ static int usb_phy_enable(int index, struct usb_ehci *ehci)
 		return ret;
 
 #if !defined(CONFIG_TARGET_S32G2XXAEVB) && \
-	!defined(CONFIG_TARGET_S32G3XXAEVB)
+	!defined(CONFIG_TARGET_S32G3XXAEVB) && \
+	!defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 	/* Reset USBPHY module */
 	setbits_le32(phy_ctrl, USBPHY_CTRL_SFTRST);
 	udelay(10);
@@ -282,7 +285,8 @@ int usb_phy_mode(int port)
 static void usb_oc_config(int index)
 {
 #if !defined(CONFIG_TARGET_S32G2XXAEVB) && \
-	!defined(CONFIG_TARGET_S32G3XXAEVB)
+	!defined(CONFIG_TARGET_S32G3XXAEVB) && \
+	!defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 #if defined(CONFIG_MX6)
 	struct usbnc_regs *usbnc = (struct usbnc_regs *)(USB_BASE_ADDR +
 			USB_OTHERREGS_OFFSET);
@@ -360,7 +364,8 @@ int __weak board_ehci_power(int port, int on)
 int ehci_mx6_common_init(struct usb_ehci *ehci, int index)
 {
 #if !defined(CONFIG_TARGET_S32G2XXAEVB) && \
-	!defined(CONFIG_TARGET_S32G3XXAEVB)
+	!defined(CONFIG_TARGET_S32G3XXAEVB) && \
+	!defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 	int ret;
 
 	enable_usboh3_clk(1);
@@ -380,7 +385,8 @@ int ehci_mx6_common_init(struct usb_ehci *ehci, int index)
 #endif
 
 #if defined(CONFIG_MX6) || defined(CONFIG_TARGET_S32G2XXAEVB) || \
-	defined(CONFIG_TARGET_S32G3XXAEVB)
+	defined(CONFIG_TARGET_S32G3XXAEVB) || \
+	defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 	usb_phy_enable(index, ehci);
 #endif
 
@@ -487,7 +493,8 @@ static const struct ehci_ops mx6_ehci_ops = {
 static int ehci_usb_phy_mode(struct udevice *dev)
 {
 #if !defined(CONFIG_TARGET_S32G2XXAEVB) && \
-	!defined(CONFIG_TARGET_S32G3XXAEVB)
+	!defined(CONFIG_TARGET_S32G3XXAEVB) && \
+	!defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 	struct usb_platdata *plat = dev_get_platdata(dev);
 	void *__iomem addr = (void *__iomem)devfdt_get_addr(dev);
 	void *__iomem phy_ctrl, *__iomem phy_status;
@@ -560,7 +567,8 @@ static int ehci_usb_ofdata_to_platdata(struct udevice *dev)
 static int ehci_usb_bind(struct udevice *dev)
 {
 #if !defined(CONFIG_TARGET_S32G2XXAEVB) && \
-	!defined(CONFIG_TARGET_S32G3XXAEVB)
+	!defined(CONFIG_TARGET_S32G3XXAEVB) && \
+	!defined(CONFIG_TARGET_APTIV_CVC_SOUSA)
 	/*
 	 * TODO:
 	 * This driver is only partly converted to DT probing and still uses
-- 
2.25.1

